src/components/Header.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { ErrorMessages } from '../types/ErrorMessages';
+import { USER_ID, addTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { useTodos } from '../utils/TodoContext';
+import classNames from 'classnames';
+
+export const Header: React.FC = () => {
+  const {
+    isLoading,
+    setIsLoading,
+    setTodos,
+    tempTodo,
+    setTempTodo,
+    title,
+    setTitle,
+    showError,
+    todos,
+  } = useTodos();
+
+  const userId = USER_ID;
+  const isCompleted = false;
+
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      showError(ErrorMessages.EmptyTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    setTempTodo({
+      id: 0,
+      title: title.trim(),
+      userId: USER_ID,
+      completed: false,
+    });
+
+    if (tempTodo) {
+      setTodos(prevTodos => [...prevTodos, tempTodo]);
+    }
+
+    return addTodo({ title, userId, completed: isCompleted })
```

Comment:
```suggestion
    const newTodo = {
      title: title.trim(),
      userId: USER_ID,
      completed: false,
    };
    
    setTempTodo({ ...newTodo,  id: 0 });

    addTodo(newTodo)
```
- You don't need to set temp todo to the todos array
- DRY